de.metas.procurement.base/src/main/java/de/metas/procurement/base/order/impl/PurchaseCandidate.java

```diff
@@ -133,7 +136,9 @@ private int getC_UOM_ID()
 
 	public int getM_HU_PI_Item_Product_ID()
 	{
-		return model.getM_HU_PI_Item_Product_ID();
+		final I_M_HU_PI_Item_Product huPIP = Services.get(IPMMPurchaseCandidateBL.class).getM_HU_PI_Item_Product_Effective(model);
+
+		return huPIP == null ? -1 : huPIP.getM_HU_PI_Item_Product_ID();
```

Comment:
Pls introduce and use IPMMPurchaseCandidateBL.getM_HU_PI_Item_Product_Effective_ID(model) to avoid unnecessary database loads